crates/language/src/buffer.rs

```diff
@@ -3241,14 +3241,23 @@ impl BufferSnapshot {
             Point::new(prev_non_blank_row.unwrap_or(row_range.start), 0)
                 ..Point::new(row_range.end, 0),
             |row, line| {
-                if config
+                // Either the parent language, or an injected language
+                let row_language = self
+                    .language_scope_at(Point::new(row, 0))
```

Comment:
I think we can just use `language_at` here instead of `language_scope_at`. The latter is more sophisticated and not really needed in this case.